agent/stats/engine.go

```diff
@@ -835,3 +859,26 @@ func (engine *DockerStatsEngine) ContainerDockerStats(taskARN string, containerI
 
 	return containerStats, containerNetworkRateStats, nil
 }
+
+func (engine *DockerStatsEngine) GetServiceConnectStats() error {
+	engine.lock.Lock()
+	defer engine.lock.Unlock()
+
+	var wg sync.WaitGroup
+
+	for taskArn := range engine.taskToServiceConnectStats {
+		wg.Add(1)
+		task, err := engine.resolver.ResolveTaskByARN(taskArn)
+		if err != nil {
+			return errors.Errorf("stats engine: task '%s' not found", taskArn)
+		}
+		// TODO: Check if task is service-connect enabled
+		serviceConnectStats, ok := engine.taskToServiceConnectStats[taskArn]
+		if !ok {
+			return errors.Errorf("task '%s' is not registered to collect service connect metrics", taskArn)
+		}
+		go serviceConnectStats.retrieveServiceConnectStats(task, &wg)
```

Comment:
I think `retrieveServiceConnectStats` doesn't need to be aware of the wait group. Instead you can call it like

```go
go func() {
    serviceConnectStats.retrieveServiceConnectStats()
    wg.Done()
}
```